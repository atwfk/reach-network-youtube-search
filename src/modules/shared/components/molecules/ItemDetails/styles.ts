import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledItemDetails = styled.div`
  .title {
    margin-bottom: 4px;

    @media (max-width: ${SCREENS.SM}) {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      width: 200px;
    }
  }

  .stats {
    @media (min-width: ${SCREENS.SM}) {
      margin-bottom: 10px;
    }
  }

  .description {
    display: none;
    @media (min-width: ${SCREENS.SM}) {
      display: block;
    }
  }
`;
