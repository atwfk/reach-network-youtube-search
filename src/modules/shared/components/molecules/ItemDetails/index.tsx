import React from "react";
import type { FC, ReactElement } from "react";
import Title from "../../atoms/Title";
import Stats from "../../atoms/Stats";
import Description from "../../atoms/Description";
import { StyledItemDetails } from "./styles";

const ItemDetails: FC<{ title: string; stats: string[]; description: string }> = ({
  title,
  stats,
  description
}): ReactElement => {
  return (
    <StyledItemDetails>
      <Title title={title} />
      <Stats stats={stats} />
      <Description description={description} />
    </StyledItemDetails>
  );
};

export default ItemDetails;
