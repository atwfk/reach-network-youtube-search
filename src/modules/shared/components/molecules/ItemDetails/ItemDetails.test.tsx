import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import ItemDetails from ".";

afterEach(cleanup);

describe("Title ItemDetails snapshots", () => {
  it("Should render ItemDetails component with title, description, and stats", () => {
    const tree = renderer
      .create(
        <ItemDetails
          title="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
          stats={["TechWorld with Nana", "3.6M views", "a year ago"]}
          description="Full Kubernetes Tutorial | Kubernetes Course | Hands-on course with a lot of demos Become a Kubernetes Administrator - CKA: ..."
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
