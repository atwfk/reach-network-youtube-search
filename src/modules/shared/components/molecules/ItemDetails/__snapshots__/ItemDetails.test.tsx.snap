// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title ItemDetails snapshots Should render ItemDetails component with title, description, and stats 1`] = `
.c1 {
  font-weight: 400;
  font-size: 16px;
}

.c2 {
  display: block;
  color: #767272;
  font-weight: 400;
  font-size: 15px;
}

.c2 li {
  margin-right: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 li .separator {
  margin-left: 6px;
  width: 2px;
  height: 2px;
  background-color: #767272;
  display: none;
}

.c3 {
  color: #767272;
  font-size: 15px;
}

.c0 .title {
  margin-bottom: 4px;
}

.c0 .description {
  display: none;
}

@media (min-width:768px) {
  .c1 {
    font-size: 25px;
  }
}

@media (min-width:640px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:640px) {
  .c2 li .separator {
    display: block;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 20px;
  }
}

@media (max-width:640px) {
  .c0 .title {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    width: 200px;
  }
}

@media (min-width:640px) {
  .c0 .stats {
    margin-bottom: 10px;
  }
}

@media (min-width:640px) {
  .c0 .description {
    display: block;
  }
}

<div
  className="c0"
>
  <h3
    className="c1 title"
    title="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
  >
    Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]
  </h3>
  <ul
    className="c2 stats"
  >
    <li
      data-testid="stat"
    >
      TechWorld with Nana
       
      <span
        className="separator"
      />
    </li>
    <li
      data-testid="stat"
    >
      3.6M views
       
      <span
        className="separator"
      />
    </li>
    <li
      data-testid="stat"
    >
      a year ago
       
    </li>
  </ul>
  <p
    className="c3 description"
  >
    Full Kubernetes Tutorial | Kubernetes Course | Hands-on course with a lot of demos Become a Kubernetes Administrator - CKA: ...
  </p>
</div>
`;
